addons/website/static/src/snippets/s_carousel_intro/options.js

```diff
@@ -5,22 +5,32 @@ options.registry.CarouselIntro = options.registry.Carousel.extend({
     /**
      * @override
      */
-    async _computeWidgetState(methodName, params) {
+    async selectClass(previewMode, widgetValue, params) {
         // Prevent the "Controllers" option from being "centered" when
-        // arrows and indicators are displayed.
-        if (methodName === "selectClass" && params.name === "carousel_controllers_centered_opt") {
-            const controllersEl = this.$target[0];
-            const carouselEl = controllersEl.closest(".carousel");
-            const indicatorsEl = controllersEl.querySelector(".carousel-indicators");
-            if (
-                !carouselEl.classList.contains("s_carousel_arrows_hidden")
-                && !indicatorsEl.classList.contains("s_carousel_indicators_hidden")
-            )
-            {
-                controllersEl.classList.toggle("justify-content-center");
-                controllersEl.classList.toggle("justify-content-between");
+        // arrows and indicators are displayed
+        await this._super(...arguments);
+        const isArrowsOption = [
+            "s_carousel_default",
+            "s_carousel_boxed",
+            "s_carousel_rounded",
+            "s_carousel_arrows_hidden",
+        ].includes(widgetValue);
+        const isIndicatorsOption = params.applyTo === ".carousel-indicators";
+        if (isIndicatorsOption || isArrowsOption) {
```

Comment:
I feel like this would be clearer without any of this pre-check: on any class change (whether or not it is about the arrows option or the indicator option or anything else), check if the centered value should not be removed. This will make the code way clearer IMO but will also allow for customizations to have better chances at working (here this is very hardcoded to work only with the current 4 arrow values, etc).